### PR TITLE
Add GRS arguments for pi_instance boot volume replication

### PR DESCRIPTION
**Pull request by Alexander-Kita**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPIInstanceDeploymentGRS
--- PASS: TestAccIBMPIInstanceDeploymentGRS (1164.25s)
PASS
```

Description:
- Added argument pi_boot_volume_replication_enabled (Boolean) which specifies whether the boot volume will be replication enabled
- Added argument pi_replication_sites (List) which specifies replication sites for boot volume replication. Argument above must be true to use this.
- Add GRS test to see if created boot volume replication and site is set correctly

### CODE REVIEW

**Review by ismirlia — APPROVED** [review]

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 1]
```diff
@@ -1333,18 +1347,33 @@ func createSAPInstance(d *schema.ResourceData, sapClient *st.IBMPISAPInstanceCli
...
+	}
+	var replicationSites []string
+	if sites, ok := d.GetOk(Arg_ReplicationSites); ok {
+		if !bootVolumeReplicationEnabled {
+			return nil, fmt.Errorf("must set %s to true in order to specify replication sites", Arg_BootVolumeReplicationEnabled)
+		} else {
+			replicationSites = flex.ExpandStringList(sites.([]interface{}))
+		}
+	}
+
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_instance.go:1362]: Could you set to only set those when they are present? By putting them below `body := &models....`
Using `body.BootVolumeReplicationEnabled`
Same for replicationSites

**Review by michaelkad — COMMENTED** [review]
One comment

File: ibm/service/power/resource_ibm_pi_instance_test.go [hunk 2]
```diff
@@ -765,3 +765,61 @@ func TestAccIBMPIInstanceDeploymentTypeNoStorage(t *testing.T) {
...
+	instanceRes := "ibm_pi_instance.power_instance"
+	bootVolumeData := "data.ibm_pi_volume.power_boot_volume_data"
+	name := fmt.Sprintf("tf-pi-instance-%d", acctest.RandIntRange(10, 100))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMPIInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccIBMPIInstanceGRSConfig(name, helpers.PIInstanceHealthOk, "2", "0.25"),
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_instance_test.go:779]: There should be a `OK` in the constant file. Could you replace it with that?

**Review by michaelkad — APPROVED** [review]

**Review by ismirlia — APPROVED** [review]

**Comment by Alexander-Kita** [comment]
@michaelkad Re-requesting review for changes made mentioned in #150 that were present here.

**Review by michaelkad — APPROVED** [review]

**Comment by Alexander-Kita** [comment]
Opened in main repo.

**Pull request closed without merging**